### PR TITLE
Qt: Clean up depracation warnings

### PR DESCRIPTION
**Pull request by MSuih**
Silences the following warnings:

```
warning C4996: 'QDesktopWidget::availableGeometry': Use QGuiApplication::screens()
warning C4996: 'QDesktopWidget::screenGeometry': Use QGuiApplication::screens()
warning C4996: 'QListWidgetItem::setTextColor': Use QListWidgetItem::setForeground() instead
warning C4996: 'QListWidgetItem::setBackgroundColor': Use QListWidgetItem::setBackground() instead
warning C4996: 'qSort': Use std::sort
warning C4996: 'qGreater<QModelIndex>::operator ()': Use std::greater
```

### CODE REVIEW

File: rpcs3/rpcs3qt/settings_dialog.cpp [hunk 1]
```diff
@@ -1187,12 +1188,12 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> guiSettings, std:
 		});
 		connect(ui->gs_width, &QSpinBox::editingFinished, [=]()
 		{
-			ui->gs_width->setValue(std::min(ui->gs_width->value(), QApplication::desktop()->screenGeometry().width()));
+			ui->gs_width->setValue(std::min(ui->gs_width->value(), QGuiApplication::primaryScreen()->size().width()));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/settings_dialog.cpp]: why do you use size here and geometry above?

File: rpcs3/rpcs3qt/user_manager_dialog.cpp [hunk 2]
```diff
@@ -209,7 +210,7 @@ void user_manager_dialog::UpdateTable(bool mark_only)
 		m_table->horizontalHeader()->height() + m_table->verticalHeader()->length() + m_table->frameWidth() * 2);
 
 	QSize preferred_size = minimumSize().expandedTo(sizeHint() - m_table->sizeHint() + table_size).expandedTo(size());
-	QSize max_size = QSize(preferred_size.width(), static_cast<int>(QApplication::desktop()->screenGeometry().height()*.6));
+	QSize max_size = QSize(preferred_size.width(), static_cast<int>(QGuiApplication::primaryScreen()->size().height() * 0.6));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/user_manager_dialog.cpp]: you kinda switch between geometry and size randomly between files.

File: rpcs3/rpcs3qt/settings_dialog.cpp (hunk 1 above)
Comment (MSuih) [reply rpcs3/rpcs3qt/settings_dialog.cpp]: I want to keep the line length reasonable, and geometry is twice as long. The difference between size and geometry is that geometry has a starting point, they have identical width and height.

File: rpcs3/rpcs3qt/user_manager_dialog.cpp (hunk 2 above)
Comment (MSuih) [reply rpcs3/rpcs3qt/user_manager_dialog.cpp]: Same reason as above.

File: rpcs3/rpcs3qt/settings_dialog.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/rpcs3qt/settings_dialog.cpp]: Too tired to think about all of this.
I'll just trust that everything just works

**Pull request merged**